library/std/src/net/tcp/tests.rs

```diff
@@ -46,6 +46,24 @@ fn connect_error() {
     }
 }
 
+// try to connect to an unreachable address
+#[test]
+fn connect_timeout_ok() {
+    let result =
+        TcpStream::connect_timeout(&format!("8.8.8.8:9999").parse().unwrap(), Duration::MAX);
+    assert!(result.is_ok());
```

Comment:
@eval-exec I'm nut sure what you are trying to do with this test^^. I'd expect it to never finish even with the fix. That doesn't seem right.

Reply:
> @eval-exec I'm not sure what you are trying to do with this test^^. I'd expect it to never finish even with the fix. That doesn't seem right.

I'll remove this.